analysis/src/main/java/org/hps/analysis/tuple/TupleDriver.java

```diff
@@ -604,6 +609,19 @@ protected TrackState fillParticleVariables(EventHeader event, ReconstructedParti
             
             Track track = particle.getTracks().get(0);
             TrackState trackState = track.getTrackStates().get(0);
+            Hep3Vector extrapTrackPosL0;
+            Hep3Vector extrapTrackPosL1;
+            Hep3Vector extrapTrackPosL2;
+            if(trackState.getTanLambda() > 0){
+                extrapTrackPosL0 = TrackUtils.extrapolateTrack(track,topPos[0]);
```

Comment:
In the future, you should switch to using the extrapolator which makes use of the field map.  The analytic extrapolator should be fine for L1-L2 but I'm not certain it will give the most accurate results for L0.